Code/Tools/AssetProcessor/Platform/Linux/native/FileWatcher/FileWatcher_linux.cpp

```diff
@@ -20,150 +23,151 @@ static constexpr size_t s_iNotifyMaxEntries = 1024 * 16;         // Control the
+
+            AZStd::fixed_string<255> errorString;
+            errorString.resize_no_construct(errorString.capacity());
+            if (m_iNotifyHandle < 0)
+            {
+                AZ_Warning("FileWatcher", false, "Unable to initialize inotify, file monitoring will not be available: %s\n", strerror_r(err, errorString.data(), errorString.size()));
+            }
         }
-        return (m_iNotifyHandle >= 0);
     }
+    return (m_iNotifyHandle >= 0);
+}
 
-    void Finalize()
+void FileWatcher::PlatformImplementation::Finalize()
+{
+    if (m_iNotifyHandle >= 0)
     {
-        if (m_iNotifyHandle >= 0)
+        if (!m_handleToFolderMapLock.tryLock(s_handleToFolderMapLockTimeout))
         {
-            if (!m_handleToFolderMapLock.tryLock(s_handleToFolderMapLockTimeout))
-            {
-                AZ_Error("FileWatcher", false, "Unable to obtain inotify handle lock on thread");
-                return;
-            }
-
-            QHashIterator<int, QString> iter(m_handleToFolderMap);
-            while (iter.hasNext())
-            {
-                iter.next();
-                int watchHandle = iter.key();
-                inotify_rm_watch(m_iNotifyHandle, watchHandle);
-            }
-            m_handleToFolderMap.clear();
-            m_handleToFolderMapLock.unlock();
+            AZ_Error("FileWatcher", false, "Unable to obtain inotify handle lock on thread");
+            return;
+        }
 
-            ::close(m_iNotifyHandle);
-            m_iNotifyHandle = -1;
+        QHashIterator<int, QString> iter(m_handleToFolderMap);
+        while (iter.hasNext())
+        {
+            iter.next();
+            int watchHandle = iter.key();
+            inotify_rm_watch(m_iNotifyHandle, watchHandle);
         }
+        m_handleToFolderMap.clear();
+        m_handleToFolderMapLock.unlock();
+
+        ::close(m_iNotifyHandle);
+        m_iNotifyHandle = -1;
     }
+}
 
-    void AddWatchFolder(QString folder)
+void FileWatcher::PlatformImplementation::AddWatchFolder(QString folder, bool recursive)
+{
+    if (m_iNotifyHandle < 0)
     {
-        if (m_iNotifyHandle >= 0)
-        {
-            // Clean up the path before accepting it as a watch folder
-            QString cleanPath = QDir::cleanPath(folder);
-
-            // Add the folder to watch and track it
-            int watchHandle = inotify_add_watch(m_iNotifyHandle, 
-                                                cleanPath.toUtf8().constData(),
-                                                IN_CREATE | IN_CLOSE_WRITE | IN_DELETE | IN_DELETE_SELF | IN_MODIFY | IN_MOVE);
-            
-            if (!m_handleToFolderMapLock.tryLock(s_handleToFolderMapLockTimeout))
-            {
-                AZ_Error("FileWatcher", false, "Unable to obtain inotify handle lock on thread");
-                return;
-            }
-            m_handleToFolderMap[watchHandle] = cleanPath;
-            m_handleToFolderMapLock.unlock();
+        return;
```

Comment:
Warning or an assert would be useful here

Reply:
We've already told them that inotify wasn't working, do we need to tell them that every path they add also won't work?